docs/advanced_topics/i18n/duplicate_tree.rst

```diff
@@ -46,6 +54,22 @@ We must set Django's ``LANGUAGES`` setting so we don't redirect non English/Fren
 
             return HttpResponseRedirect(self.url + language + '/')
 
+    # your-app.middleware.py
+    from django.utils import translation
+    from django.utils.translation import get_language_from_request
+
+    class LocaleMiddleware(object):
```

Comment:
How does this differ from Django's builtin ``LocaleMiddleware``?

https://docs.djangoproject.com/en/1.11/ref/middleware/#module-django.middleware.locale